qutebrowser/misc/sql.py

```diff
@@ -0,0 +1,208 @@
+    result.next()
+    return result.record().value(0)
+
+
+def _prepare_query(querystr):
+    log.sql.debug('Preparing SQL query: "{}"'.format(querystr))
+    database = QSqlDatabase.database()
+    query = QSqlQuery(database)
+    query.prepare(querystr)
+    return query
+
+
+def run_query(querystr, values=None):
+    """Run the given SQL query string on the database.
+
+    Args:
+        values: A list of positional parameter bindings.
+    """
+    query = _prepare_query(querystr)
+    for val in values or []:
+        query.addBindValue(val)
+    log.sql.debug('Query bindings: {}'.format(query.boundValues()))
+    if not query.exec_():
+        raise SqlException('Failed to exec query "{}": "{}"'.format(
+                           querystr, query.lastError().text()))
+    return query
+
+
+def run_batch(querystr, values):
+    """Run the given SQL query string on the database in batch mode.
+
+    Args:
+        values: A list of lists, where each inner list contains positional
+                bindings for one run of the batch.
+    """
+    query = _prepare_query(querystr)
+    transposed = [list(row) for row in zip(*values)]
+    for val in transposed:
+        query.addBindValue(val)
+    log.sql.debug('Batch Query bindings: {}'.format(query.boundValues()))
+
+    db = QSqlDatabase.database()
+    db.transaction()
+    if not query.execBatch():
+        raise SqlException('Failed to exec query "{}": "{}"'.format(
+                           querystr, query.lastError().text()))
+    db.commit()
+
+    return query
+
+
+class SqlTable(QObject):
+
+    """Interface to a sql table.
+
+    Attributes:
+        Entry: The class wrapping row data from this table.
+        _name: Name of the SQL table this wraps.
+
+    Signals:
+        changed: Emitted when the table is modified.
+    """
+
+    changed = pyqtSignal()
+
+    def __init__(self, name, fields, parent=None):
+        """Create a new table in the sql database.
+
+        Raises SqlException if the table already exists.
+
+        Args:
+            name: Name of the table.
+            fields: A list of field names.
+        """
+        super().__init__(parent)
+        self._name = name
+        run_query("CREATE TABLE IF NOT EXISTS {} ({})"
+                  .format(name, ','.join(fields)))
+        # pylint: disable=invalid-name
+        self.Entry = collections.namedtuple(name + '_Entry', fields)
+
+    def __iter__(self):
+        """Iterate rows in the table."""
+        result = run_query("SELECT * FROM {}".format(self._name))
+        while result.next():
+            rec = result.record()
+            yield self.Entry(*[rec.value(i) for i in range(rec.count())])
+
+    def contains(self, field, value):
+        """Return whether the table contains the matching item.
+
+        Args:
+            field: Field to match.
+            value: Value to check for the given field.
+        """
+        query = run_query("SELECT * FROM {} where {} = ? LIMIT 1"
```

Comment:
I'd suggest using `SELECT 1` instead (so we don't actually need to fetch much data) - and if you want a boolean back, you can do `EXISTS(SELECT 1 FROM {} where {} = ? LIMIT 1)`.